packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/read.rs

```diff
@@ -56,23 +63,26 @@ where
             .change_context(QueryError)?
             .map(|row| row.into_report().change_context(QueryError))
             .and_then(|row| async move {
-                let versioned_uri = VersionedUri::from_str(row.get(versioned_uri_index))
+                let base_uri = BaseUri::new(row.get(base_uri_index))
                     .into_report()
                     .change_context(QueryError)?;
+                let version: i64 = row.get(version_index);
+                let version = OntologyTypeVersion::new(version as u32);
+                let owned_by_id = OwnedById::new(row.get(owned_by_id_index));
+                let updated_by_id = UpdatedById::new(row.get(updated_by_id_path_index));
+
                 let record_repr: <T::OntologyType as OntologyType>::Representation =
                     serde_json::from_value(row.get(schema_index))
                         .into_report()
                         .change_context(QueryError)?;
                 let record = T::OntologyType::try_from(record_repr)
                     .into_report()
                     .change_context(QueryError)?;
-                let owned_by_id = OwnedById::new(row.get(owned_by_id_index));
-                let updated_by_id = UpdatedById::new(row.get(updated_by_id_path_index));
 
                 Ok(T::new(
                     record,
                     OntologyElementMetadata::new(
-                        OntologyTypeEditionId::from(&versioned_uri),
+                        OntologyTypeEditionId::new(base_uri, version),
```

Comment:
The change made here to change the queried`versioned_uri` seems unrelated to the rest of the changes. Was this done to prevent looking into `schema->>$id` to get the versioned URI?

Reply:
Yes, I wanted to avoid the schema lookup here. We have to look in the `type_ids` table anyway, so there is no reason to not directly look them up there.